Src/AutoFixture.NUnit3/AutoDataAttribute.cs

```diff
@@ -0,0 +1,275 @@
+        /// </summary>
+        protected AutoDataAttribute(IFixture fixture)
+        {
+            _fixture = fixture;
+            this.RunState = RunState.Runnable;
+            this.Properties = new PropertyBag();
+        }
+
+        /// <summary>
+        ///     Construct one or more TestMethods from a given MethodInfo,
+        ///     using available parameter data.
+        /// </summary>
+        /// <param name="method">The MethodInfo for which tests are to be constructed.</param>
+        /// <param name="suite">The suite to which the tests will be added.</param>
+        /// <returns>One or more TestMethods</returns>
+        public IEnumerable<TestMethod> BuildFrom(IMethodInfo method, Test suite)
+        {
+            var test = new NUnitTestCaseBuilder().BuildTestMethod(method, suite, this.GetParametersForMethod(method));
+
+            yield return test;
+        }
+
+        private TestCaseParameters GetParametersForMethod(IMethodInfo method)
+        {
+            try
+            {
+                var specimenBuilder = new SpecimenContext(this._fixture);
+                var parameterValues = method.GetParameters()
+                    .Select(p => specimenBuilder.Resolve(p.ParameterType))
+                    .ToArray();
+                return new TestCaseParameters(parameterValues);
+            }
+            catch (Exception ex)
+            {
+                return new TestCaseParameters(ex);
+            }
+        }
+
+        #region ITestData Members
+
+        /// <summary>
+        ///     Gets or sets the name of the test.
+        /// </summary>
+        /// <value>The name of the test.</value>
+        public string TestName { get; set; }
+
+        /// <summary>
+        ///     Gets or sets the RunState of this test case.
+        /// </summary>
+        public RunState RunState { get; private set; }
+
+        /// <summary>
+        ///     Gets the list of arguments to a test case
+        /// </summary>
+        public object[] Arguments { get; private set; }
+
+        /// <summary>
+        ///     Gets the properties of the test case
+        /// </summary>
+        public IPropertyBag Properties { get; private set; }
+
+        #endregion
+
+        #region ITestCaseData Members
+
+        /// <summary>
+        ///     Gets or sets the expected result.
+        /// </summary>
+        /// <value>The result.</value>
+        public object ExpectedResult
+        {
+            get
+            {
+                return this._expectedResult;
+            }
+            set
+            {
+                this._expectedResult = value;
+                this.HasExpectedResult = true;
+            }
+        }
+
+        private object _expectedResult;
+
+        /// <summary>
+        ///     Returns true if the expected result has been set
+        /// </summary>
+        public bool HasExpectedResult { get; private set; }
+
+        #endregion
+
+        #region Other Properties
```

Comment:
Why are the properties in this region present on this class?

I tried to delete them all, and the build still succeeds.